drivers/gles3/storage/texture_storage.h

```diff
@@ -514,7 +512,9 @@ class TextureStorage : public RendererTextureStorage {
 	virtual void texture_3d_initialize(RID p_texture, Image::Format, int p_width, int p_height, int p_depth, bool p_mipmaps, const Vector<Ref<Image>> &p_data) override;
 	virtual void texture_proxy_initialize(RID p_texture, RID p_base) override; //all slices, then all the mipmaps, must be coherent
 
-	RID texture_create_external(Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type = RS::TEXTURE_LAYERED_2D_ARRAY);
+	virtual RID texture_create_external(Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type = RS::TEXTURE_LAYERED_2D_ARRAY);
+
+	virtual RID texture_set_external(int p_width, int p_height) override;
```

Comment:
I don't think this is a great name for this function. It's creating a new texture, but it is named like a setter.

I think we should probably call this `texture_create_external()` and then rename the existing `texture_create_external()` to something else:

```suggestion
	virtual RID texture_create_reference(Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type = RS::TEXTURE_LAYERED_2D_ARRAY);

	virtual RID texture_create_external(int p_width, int p_height) override;
```

I'm not totally sure about the name `texture_create_reference()` - this is something we should discuss.

Reply:
An alternative to using `RID texture_create_external(int p_width, int p_height)`, would be to use `void texture_external_initialize(RID p_texture, int p_width, int p_height)`? Something like that is done in PR 83519 and it matches some of the other `texture_*_initialize()` functions

Reply:
Shouldn't this PR https://github.com/godotengine/godot/pull/83519 be merge then ? I could do handle the renaming part on it, then it could be merge and focus this one on camera feed API ?

What would be the fastest way to integration ?

Reply:
If @m4gr3d has time to work on #83519, then finishing that and merging it first is an option! If not, someone else may need to make a superseding PR (which could even be this PR).

However, I personally think we should handle the renaming in whichever PR we add support for `GL_TEXTURE_EXTERNAL_OES`, so we don't have a period of time where there's confusing function names in the renderer. If we add that in #83519, I think that's where we should figure out the naming.

Reply:
This was superseeded by https://github.com/godotengine/godot/pull/96982 and merged.